cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java

```diff
@@ -181,6 +198,37 @@ public Integer doAction() throws Exception {
     return new Tuple<>(generationInputs, selectedGenerableArtifacts);
   }
 
+  /**
+   * Opens a repeatable prompt with a yes/no question and sets the output directory to the current user directory.
+   *
+   */
+  private void setOutputPathWithPrompt() {
+
+    Path outputDirectory = Paths.get(System.getProperty("user.dir"));
+    boolean decided = false;
+    while (!decided) {
+      String userInput = getUserInput();
+      switch (userInput) {
+        case "yes":
+        case "y":
+        case "":
+          LOG.info("Set output directory to: {}", outputDirectory.toString());
+          this.outputRootPath = outputDirectory;
+          decided = true;
+          break;
+        case "no":
+        case "n":
+          LOG.info("Cancelling generate process... you can use -o to explicitly set the output directory.");
+          decided = true;
+          System.exit(0);
```

Comment:
Just asking myself whether this makes trouble during testing.
I remembered that I had really issues identifying and maintaining system edits which where coming everywhere in the code. No clear return code facade was done.
If I am not totally wrong, I reimplemented that to have just two exits in the CobiGenCLI top level class or top level command.
Please check. From an architectural point of view, it would be much more convenient to throw a user abort exception here and catch it and edit with a different code to be able to distinguish between an error and user driven abortion in tests.
In devonfw IDE I think we have return code 255 for that. Never googled it but could be a standard and if not let's comply at least with devonfw ide.

Reply:
Added new UserAbortException and changed error code to 255.